### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by inductor**
This PR adds Golang CI with GitHub Actions based on the https://github.com/google/yamlfmt/issues/9 requirement.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go }}
+          cache: true
+        
+      - name: Install goimports
+        run: go get golang.org/x/tools/cmd/goimports
```
Comment (braydonk) [inline .github/workflows/ci.yaml:23]: If I'm not mistaken, go 1.18 remove the ability to use `go get` for tools? I think you might need to use `go install` here.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        run: go mod tidy
+
+      - name: Test
+        run: go test -v ./...
+
+      - name: Test
+        run: go vet
+    
+      - name: goimports
+        run: test -z "$(set -o pipefail && $(go env GOPATH)/bin/goimports -l . | tee goimports.out)" || { cat goimports.out && exit 1; }
```
Comment (braydonk) [inline .github/workflows/ci.yaml:35]: I think `setup-go@v3` actually adds GOBIN to the PATH (based on these docs https://github.com/actions/setup-go#v3)

So you can likely just call `goimports` (provided you also use `go install` as the previous comment suggested).

**Review by braydonk — COMMENTED** [review]
I appreciate the PR! Just a couple comments on how goimports is installed.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (braydonk) [reply .github/workflows/ci.yaml:23]: Go 1.18 did indeed remove the ability to use `go get` for this: https://go.dev/doc/go-get-install-deprecation
So will need to switch to `go install` most likely.

**Comment by inductor** [comment]
@braydonk Thanks for the comments. I fixed the above so PTAL!

**Review by braydonk — APPROVED** [review]
Looks good, thanks!

**Pull request merged**